.github/workflows/update.yml

```diff
@@ -1,29 +1,82 @@
 name: brew formula update
+
 on:
   workflow_dispatch:
     inputs:
-      wash_version:
-        description: "Version of wash, without the `v`, to release to homebrew"
+      tag_prefix:
+        description: "Prefix for the release tag to pull from (i.e. wash-cli)"
+        default: "wash-cli"
+        required: false
+        type: string
+      tag_version:
+        description: "Tag to pull"
         required: true
-        default: "0.18.0"
+        type: string
+
 jobs:
   update-release:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Prepare Formula
+      - uses: actions/checkout@v4
+
+      # This is needed by Goreleaser to build the placeholder Go binary.
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+
+      - name: Normalize inputs for pipeline use
+        uses: actions/github-script@v7
+        id: normalize_inputs
+        with:
+          script: |
+            const input_tag = context.payload.inputs.tag_version
+            const input_prefix = context.payload.inputs.tag_prefix
+            const tag_no_v = input_tag.replaceAll('v', '')
+            const tag_with_v = 'v' + tag_no_v
+            const prefix = input_prefix.replace(/^-+/g, '').replace(/-+$/g, '')
+            const release_name = prefix + '-' + tag_with_v
+            core.setOutput('tag_no_v', tag_no_v)
+            core.setOutput('tag_with_v', tag_with_v)
+            core.setOutput('release_name', release_name)
+
+      - name: Download release artifacts
         env:
-          WASMCLOUD_URL: https://github.com/wasmCloud/wash/archive/v${{ inputs.wash_version }}.tar.gz
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh -R wasmcloud/wasmcloud release download ${{ steps.normalize_inputs.outputs.release_name }} -D tmp/ -p '*apple-darwin*' -p '*unknown-linux*'
+
+      - name: Setup placeholder files for Goreleaser
+        shell: bash
+        run: |
+          mkdir tmp
+          cat <<EOF > tmp/main.go
+          package main
+          func main() {}
+          EOF
```

Comment:
Hm, this is go code but it doesn't contain any `if err != nil` blocks... will this even compile?